File: frontend/lib/models/parts_search_dto.dart
class PartsSearchDto {
  String? searchQuery;
  int? manufacturerId;
  String? catalogNumber;
  bool? isUnique;
  String? equipmentType;
  String? categoryCode;
  int page;
  int size;
  String sortBy;
  String sortDirection;

  PartsSearchDto({
    this.searchQuery,
    this.manufacturerId,
    this.catalogNumber,
    this.isUnique,
    this.equipmentType,
    this.categoryCode,
    this.page = 0,
    this.size = 20,
    this.sortBy = 'catalogId',
    this.sortDirection = 'ASC',
  });

  Map<String, dynamic> toJson() => {
        'searchQuery': searchQuery,
        'manufacturerId': manufacturerId,
        'catalogNumber': catalogNumber,
        'isUnique': isUnique,
        'equipmentType': equipmentType,
        'categoryCode': categoryCode,
        'page': page,
        'size': size,
        'sortBy': sortBy,
        'sortDirection': sortDirection,
      };
}

File: backend/src/main/java/ru/bowling/bowlingapp/DTO/PartsSearchDTO.java
package ru.bowling.bowlingapp.DTO;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PartsSearchDTO {

    private String searchQuery;
    private Long manufacturerId;
    private String catalogNumber;
    private Boolean isUnique;
    private String equipmentType;
    private String categoryCode;
    private Long componentId;
    
    @Builder.Default
    private Integer page = 0;
    @Builder.Default
    private Integer size = 20;
    @Builder.Default
    private String sortBy = "catalog_id";
    @Builder.Default
    private String sortDirection = "ASC";
}
